### PR TITLE
Add communication between simulator and native firmware

### PR DESCRIPTION
**Pull request by ecourtois**
Use a shared memory array containing sensors values provided by the simulator.
Also add pose order info in output of "_pose" shell command.

### CODE REVIEW

**Review by gdoffe — CHANGES_REQUESTED** [review]
Remove 'Fix' at the end of the commit message

**Review by gdoffe — CHANGES_REQUESTED** [review]
platforms: cortex: add "_set_shm_key" command

File: boards/cogip2019-cortex-native/vl53l0x.c [hunk 1]
```diff
@@ -49,8 +53,20 @@ void vl53l0x_init(void)
...
 uint16_t vl53l0x_continuous_ranging_get_measure(vl53l0x_t dev)
 {
-    (void) dev;
+    /* Try to initialize shared memory if not already done */
+    if(shm_ptr == NULL && pf_shm_key != 0) {
+        int shmid = shmget(pf_shm_key, VL53L0X_NUMOF*sizeof(uint16_t), 0);
+        shm_ptr = (uint16_t*) shmat(shmid,(void*)0,0);
+    }
+
+    /* Return max value is shared memory not initialized */
```
Comment (gdoffe) [inline boards/cogip2019-cortex-native/vl53l0x.c:62]:
```suggestion
    /* Return max value if shared memory is not initialized */
```

**Review by gdoffe — CHANGES_REQUESTED** [review]
Attach to **a** shared memory array

**Review by gdoffe — COMMENTED** [review]
shell: platform: add pose order to "_pose" command

Add pose order information to the output of "_pose" command.
Used by the simulator to display the current pose order.

**Review by gdoffe — CHANGES_REQUESTED** [review]
**platforms: cortex:** add pose order to "_pose" command

Add pose order information**s** to the output of "_pose" command.
Used by the simulator to display the current pose order.

File: boards/cogip2019-cortex-native/vl53l0x.c (hunk 1 above)
Comment (ecourtois) [reply boards/cogip2019-cortex-native/vl53l0x.c:62]: Attach to **the** shared memory array

**Review by gdoffe — APPROVED** [review]

**Pull request merged**